api/chat.go

```diff
@@ -68,6 +76,106 @@ func configGinChatRouter(router *gin.RouterGroup) {
 	})
 }
 
+func handleSubmitPollOptionVote(ctx tools.TUMLiveContext, msg []byte) {
+	var req submitPollOptionVote
+	if err := json.Unmarshal(msg, &req); err != nil {
+		log.WithError(err).Warn("could not unmarshal submit poll answer request")
+		return
+	}
+	if ctx.User == nil {
+		return
+	}
+
+	if err := dao.AddChatPollOptionVote(req.PollOptionId, ctx.User.ID); err != nil {
+		return
+	}
+
+	voteCount, _ := dao.GetPollOptionVoteCount(req.PollOptionId)
+
+	voteUpdateMap := gin.H{
+		"pollOptionId": req.PollOptionId,
+		"votes":        voteCount,
+	}
+
+	if voteUpdateJson, err := json.Marshal(voteUpdateMap); err == nil {
+		broadcastStream(ctx.Stream.ID, voteUpdateJson)
+	}
+}
+
+func handleStartPoll(ctx tools.TUMLiveContext, msg []byte) {
+	var req startPollReq
+	if err := json.Unmarshal(msg, &req); err != nil {
+		log.WithError(err).Warn("could not unmarshal start poll request")
+		return
+	}
+	if ctx.User == nil || !ctx.User.IsAdminOfCourse(*ctx.Course) {
+		return
+	}
+
+	var pollOptions []model.PollOption
+	for _, answer := range req.PollAnswers {
+		pollOptions = append(pollOptions, model.PollOption{
+			Answer: answer,
+		})
+	}
+
+	poll := model.Poll{
+		StreamID:    ctx.Stream.ID,
+		Question:    req.Question,
+		Active:      true,
+		PollOptions: pollOptions,
+	}
+
+	if err := dao.AddChatPoll(&poll); err != nil {
+		return
+	}
+
+	var pollOptionsJson []gin.H
+	for _, option := range poll.PollOptions {
+		pollOptionsJson = append(pollOptionsJson, option.GetStatsMap(0))
+	}
+
+	pollMap := gin.H{
+		"active":      true,
+		"question":    poll.Question,
+		"pollOptions": pollOptionsJson,
+		"submitted":   0,
+	}
```

Comment:
Using gin.H instead of declaring a type is an interesting choice, fine by me though :)